### PR TITLE
add bulk import plugin

### PR DESCRIPTION
**Pull request by debsmita1**
**Description**
- Adds the Bulk import plugins in the showcase application

**Fixes:**
https://issues.redhat.com/browse/RHIDP-2738

**Screenshot:**
Screenshot 2024-08-29 at 9 46 33 AM

**Test setup:**

- app-config changes
```
dynamicPlugins:
  rootDirectory: dynamic-plugins-root
  frontend:
    janus-idp.backstage-plugin-bulk-import:
      appIcons:
        - name: bulkImportIcon
          module: BulkImportPlugin
          importName: BulkImportIcon
      dynamicRoutes:
        - path: /bulk-import/repositories
          importName: BulkImportPage
          module: BulkImportPlugin
          menuItem:
            icon: bulkImportIcon
            text: Bulk import
```

### CODE REVIEW

File: dynamic-plugins.default.yaml [hunk 1]
```diff
@@ -475,6 +475,27 @@ plugins:
                     - isKind: resource
                     - isType: kubernetes-cluster
 
+  # Group: Bulk import
+  - package: ./dynamic-plugins/dist/janus-idp-backstage-plugin-bulk-import-backend-dynamic
+    disabled: true
```
Comment (invincibleJai) [inline dynamic-plugins.default.yaml:480]: @debsmita1 no additional config is needed for the backend plugin?

**Review by invincibleJai — COMMENTED** [review]
Looks good, needs to verify it once

File: dynamic-plugins.default.yaml (hunk 1 above)
Comment (debsmita1) [reply dynamic-plugins.default.yaml:480]: no additional config is needed for the backend plugin

**Comment by invincibleJai** [comment]
/approve

@ciiay PTAL and LGTM if all looks good

**Review by rm3l — COMMENTED** [review]
I just tested the `quay.io/janus-idp/backstage-showcase:pr-1541` on a cluster using the Operator, and can confirm this works.

- By default, the bulk import plugins are disabled:

<details>
<summary>init container logs</summary>

```
======= Skipping disabled dynamic plugin ./dynamic-plugins/dist/janus-idp-backstage-plugin-bulk-import-backend-dynamic

======= Skipping disabled dynamic plugin ./dynamic-plugins/dist/janus-idp-backstage-plugin-bulk-import
```

</details>

<details>
<summary>No "Bulk Import" menu item</summary>

image


</details>


- After enabling both the backend and frontend plugins, we can see the bulk import plugins handled by the init container:

<details>
<summary>init container logs</summary>

```
[...]

======= Installing dynamic plugin ./dynamic-plugins/dist/janus-idp-backstage-plugin-bulk-import-backend-dynamic
        ==> Grabbing package archive through `npm pack`
        ==> Removing previous plugin directory /dynamic-plugins-root/janus-idp-backstage-plugin-bulk-import-backend-dynamic-1.3.0
        ==> Extracting package archive /dynamic-plugins-root/janus-idp-backstage-plugin-bulk-import-backend-dynamic-1.3.0.tgz
        ==> Removing package archive /dynamic-plugins-root/janus-idp-backstage-plugin-bulk-import-backend-dynamic-1.3.0.tgz
        ==> Successfully installed dynamic plugin /opt/app-root/src/dynamic-plugins/dist/janus-idp-backstage-plugin-bulk-import-backend-dynamic

======= Installing dynamic plugin ./dynamic-plugins/dist/janus-idp-backstage-plugin-bulk-import
        ==> Grabbing package archive through `npm pack`
        ==> Removing previous plugin directory /dynamic-plugins-root/janus-idp-backstage-plugin-bulk-import-1.4.1
        ==> Extracting package archive /dynamic-plugins-root/janus-idp-backstage-plugin-bulk-import-1.4.1.tgz
        ==> Removing package archive /dynamic-plugins-root/janus-idp-backstage-plugin-bulk-import-1.4.1.tgz
        ==> Merging plugin-specific configuration
        ==> Successfully installed dynamic plugin /opt/app-root/src/dynamic-plugins/dist/janus-idp-backstage-plugin-bulk-import
[...]
```

</details>


I have a few questions after enabling the plugins; I guess these can be tackled as separate issues if they make sense:

- when logging in as guest (with `permission.enabled` set to `true`, the `Bulk Import` menu shows up but clicking on it returns an error page. Is that expected? Unless I missed something, I remember that it used to be hidden for users without the right permission, no? (at least last time I tested https://github.com/janus-idp/backstage-plugins/pull/2034 from the plugins repo:

image

- once logged in (via GitHub), it looks like there are issues with the dark theme:

image

- text is displayed properly with the light theme:

image

image

**Comment by debsmita1** [comment]
Thank you @rm3l  for testing this
> * when logging in as guest (with `permission.enabled` set to `true`, the `Bulk Import` menu shows up but clicking on it returns an error page. Is that expected? Unless I missed something, I remember that it used to be hidden for users without the right permission, no? (at least last time I tested [feat(bulk-import): add permission checks in bulk import UI backstage-plugins#2034](https://github.com/janus-idp/backstage-plugins/pull/2034) from the plugins repo:
> 
> image
> 

It functions differently in the plugins repository because there, we add the `BulkImportSidebarItem`, which only appears in the navigation when the user is authorized to access the plugin. However, in the showcase repository, where it's configured as a dynamic plugin, we include the bulk import icon directly in the menuItem like this:

```
menuItem:
     icon: bulkImportIcon
     text: Bulk import
```

This difference is why, even when a user isn't authorized, they can still see the icon, but they cannot access the Bulk import page since it includes access checks.


> * once logged in (via GitHub), it looks like there are issues with the dark theme:
> 
> image
> 

I can address this in a separate ticket

**Comment by rm3l** [comment]
> Thank you @rm3l for testing this
> 
> > * when logging in as guest (with `permission.enabled` set to `true`, the `Bulk Import` menu shows up but clicking on it returns an error page. Is that expected? Unless I missed something, I remember that it used to be hidden for users without the right permission, no? (at least last time I tested [feat(bulk-import): add permission checks in bulk import UI backstage-plugins#2034](https://github.com/janus-idp/backstage-plugins/pull/2034) from the plugins repo:
> > 
> > image
> 
> It functions differently in the plugins repository because there, we add the `BulkImportSidebarItem`, which only appears in the navigation when the user is authorized to access the plugin. However, in the showcase repository, where it's configured as a dynamic plugin, we include the bulk import icon directly in the menuItem like this:
> 
> ```
> menuItem:
>      icon: bulkImportIcon
>      text: Bulk import
> ```
> 
> This difference is why, even when a user isn't authorized, they can still see the icon, but they cannot access the Bulk import page since it includes access checks.
> 

Okay, thanks for the explanations.

> > * once logged in (via GitHub), it looks like there are issues with the dark theme:
> > 
> > image
> 
> I can address this in a separate ticket

Sounds good. Thanks @debsmita1 !

/lgtm

**Comment by rm3l** [comment]
@debsmita1 Looks like there are Git conflicts. Could you please rebase it? Thanks.

**Comment by rm3l** [comment]
> @rm3l: #1541 failed to apply on top of branch "1.3.x":
> 
> ```
> Applying: add bulk import plugin
> Using index info to reconstruct a base tree...
> M	dynamic-plugins.default.yaml
> M	dynamic-plugins/imports/package.json
> Falling back to patching base and 3-way merge...
> Auto-merging dynamic-plugins/imports/package.json
> CONFLICT (content): Merge conflict in dynamic-plugins/imports/package.json
> Auto-merging dynamic-plugins.default.yaml
> error: Failed to merge in the changes.
> hint: Use 'git am --show-current-patch=diff' to see the failed patch
> Patch failed at 0001 add bulk import plugin
> When you have resolved this problem, run "git am --continue".
> If you prefer to skip this patch, run "git am --skip" instead.
> To restore the original branch and stop patching, run "git am --abort".
> ```

Looking into this.. Cherry-picking it manually..

**Pull request merged**